core/src/main/java/com/linecorp/armeria/server/misc/Inet4AddressBlock.java

```diff
@@ -0,0 +1,137 @@
+
+    private static final byte[] localhost = { 127, 0, 0, 1 };
+
+    private final Inet4Address baseAddress;
+    private final int maskBits;
+    private final int lowerBound;
+    private final int upperBound;
+
+    Inet4AddressBlock(Inet4Address baseAddress, int maskBits) {
+        this.baseAddress = requireNonNull(baseAddress, "baseAddress");
+        checkArgument(maskBits >= 0 && maskBits <= 32,
+                      "maskBits: %s (expected: 0-32)", maskBits);
+        this.maskBits = maskBits;
+
+        if (maskBits == 32) {
+            lowerBound = upperBound = ipv4AddressToInt(baseAddress.getAddress());
+        } else if (maskBits == 0) {
+            lowerBound = upperBound = 0;
+        } else {
+            // Calculate the lower and upper bounds of this address block.
+            // e.g. mask is 256(0x00000100) if maskBits is 24.
+            final int mask = 1 << 32 - maskBits;
+            // e.g. (-mask) is 0xFFFFFF00 if mask is 256.
+            lowerBound = ipv4AddressToInt(baseAddress.getAddress()) & -mask;
+            // e.g. (mask-1) is 0x000000FF if mask is 256.
+            upperBound = lowerBound + mask - 1;
+        }
+    }
+
+    @Override
+    public boolean test(InetAddress address) {
+        requireNonNull(address, "address");
+        if (maskBits == 0) {
+            return true;
+        }
+        try {
+            final int addr = inetAddressToInt(address);
+            return addr >= lowerBound && addr <= upperBound;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    @Override
+    public String toString() {
+        return MoreObjects.toStringHelper(this)
+                          .add("baseAddress", baseAddress)
+                          .add("maskBits", maskBits)
+                          .add("lowerBound", "0x" + Integer.toHexString(lowerBound))
+                          .add("upperBound", "0x" + Integer.toHexString(upperBound))
+                          .toString();
+    }
+
+    /**
+     * Returns the integer representation of the specified {@link InetAddress}.
+     */
+    private static int inetAddressToInt(InetAddress inetAddress) {
+        final byte[] address;
+        if (inetAddress instanceof Inet6Address) {
+            address = ipv6ToIpv4Address((Inet6Address) inetAddress);
+        } else {
+            address = inetAddress.getAddress();
+        }
+        return ipv4AddressToInt(address);
+    }
+
+    /**
+     * Returns IPv4 byte representation of the specified {@link Inet6Address}.
+     *
+     * @throws IllegalArgumentException if the IPv6 cannot be mapped to IPv4
+     */
+    private static byte[] ipv6ToIpv4Address(Inet6Address address) {
+        final byte[] addr = address.getAddress();
+        // The first 10 bytes should be 0.
+        for (int i = 0; i < 10; i++) {
+            if (addr[i] != 0x00) {
+                throw new IllegalArgumentException("This IPv6 address cannot be used in IPv4 context" +
```

Comment:
`context: "`? so that the message is read easily?

Reply:
I missed that. Thanks!